CRM/Event/BAO/Participant.php

```diff
@@ -1904,4 +1904,36 @@ public static function getSelfServiceEligibility(int $participantId, string $url
     return $details;
   }
 
+  /**
+   * Callback for hook_civicrm_pre().
+   * @param \Civi\Core\Event\PreEvent $event
+   * @throws CRM_Core_Exception
+   */
+  public static function self_hook_civicrm_pre(\Civi\Core\Event\PreEvent $event) {
+    if ($event->entity === 'Participant' && $event->action === 'create' && !array_key_exists('created_id', $event->params)) {
```

Comment:
looks good but perhaps this should be `!empty` since I don't think we are encouraging NULL

Reply:
@eileenmcnaughton Makes sense. I've updated the PR to check if the field is `empty()` so that we set the field if it has not been set, or is set to NULL.